RHDEVDOCS-5912 - Updates to the argo cd CLI installation and reference sections

Srivaralakshmi opened the pull request:
**Jira issue:** [RHDEVDOCS-5912](https://issues.redhat.com/browse/RHDEVDOCS-5912)

**Aligned team:** DevTools 
**Cherry-pick to versions:** `gitops-docs-1.12`

**Docs preview:**

- [Installing the GitOps CLI](https://75177--ocpdocs-pr.netlify.app/openshift-gitops/latest/installing_gitops/installing-argocd-gitops-cli)
- [GitOps argocd CLI reference](https://75177--ocpdocs-pr.netlify.app/openshift-gitops/latest/gitops_cli_argocd/argocd-gitops-cli-reference#additional-resources_argocd-gitops-cli-reference)
- [Red Hat OpenShift GitOps release notes](https://75177--ocpdocs-pr.netlify.app/openshift-gitops/latest/release_notes/gitops-release-notes)

SME review: Completed by @anandf

QE review: Completed by @varshab1210 

Peer review: 

**Additional information:** This PR is related to https://github.com/openshift/openshift-docs/pull/72634. The content was already reviewed and approved. Due to the content gateway link issue from the Engineering side, certain installation procedures were commented out during the GitOps 1.12 GA. This PR comments-in back those procedures while adding admonition and updates to the content gateway links and their steps. TY!

Srivaralakshmi commented:
Hi, @anandf and @varshab1210: PTAL at the PR and help with your review inputs. Thanks!

Srivaralakshmi commented:
@anandf I have fixed the review comments. PTAL at the revised PR and provide feedback. Thank you!

anandf commented on modules/gitops-installing-argocd-cli-on-linux.adoc:
```diff
@@ -10,35 +10,42 @@ For Linux distributions, you can download the {gitops-shortname} `argocd` CLI as
 [source,terminal]
 ----
-$ sudo mv argocd-darwin-amd64 /usr/local/bin/argocd
+$ chmod +x /usr/local/bin/argocd
```
Should have prefix `sudo `

Srivaralakshmi replied:
Addressed it.

anandf commented on modules/gitops-installing-argocd-cli-on-macos.adoc:
```diff
@@ -10,29 +10,42 @@ For macOS, you can download the {gitops-shortname} `argocd` CLI as a `tar.gz` ar
 +
+[source,terminal]
+----
+$ chmod +x /usr/local/bin/argocd
```
prefix with sudo

Srivaralakshmi replied:
Addressed it.

anandf reviewed (approved):
LGTM

adellape commented on modules/gitops-installing-argocd-cli-on-linux.adoc:
```diff
@@ -10,35 +10,42 @@ For Linux distributions, you can download the {gitops-shortname} `argocd` CLI as
+|===
+|Operating system |Architecture |Tarball
 
+|Linux |x86_64, amd64 |argocd-linux-amd64.tar.gz
+|Linux on {ibmzProductName} and {linuxoneProductName} |s390x |argocd-linux-s390x.tar.gz
+|Linux on {ibmpowerProductName} |ppc64le |argocd-linux-ppc64le.tar.gz
+|Linux on ARM |aarch64, arm64 |argocd-linux-arm64.tar.gz
```
Suggest marking up the file names:
```suggestion
|Linux |x86_64, amd64 |`argocd-linux-amd64.tar.gz`
|Linux on {ibmzProductName} and {linuxoneProductName} |s390x |`argocd-linux-s390x.tar.gz`
|Linux on {ibmpowerProductName} |ppc64le |`argocd-linux-ppc64le.tar.gz`
|Linux on ARM |aarch64, arm64 |`argocd-linux-arm64.tar.gz`
```

adellape commented on modules/gitops-installing-argocd-cli-on-linux.adoc:
```diff
@@ -10,35 +10,42 @@ For Linux distributions, you can download the {gitops-shortname} `argocd` CLI as
-// gitops-installing-argocd-cli-on-linux-using-rpm.adoc
+[NOTE]
+====
+Newer versions of the CLI tool is compatible with the older versions of {gitops-title} server but not vice-versa.
```
```suggestion
Newer versions of the CLI tool are compatible with the older versions of {gitops-title} server, but not vice versa.
```
ISG and SSG use the spelling "vice versa".

adellape commented on modules/gitops-installing-argocd-cli-on-macos.adoc:
```diff
@@ -10,29 +10,42 @@ For macOS, you can download the {gitops-shortname} `argocd` CLI as a `tar.gz` ar
+|===
+|Operating system |Architecture |Tarball
 
+|macOS on Intel |x86_64 |argocd-macos-amd64.tar.gz
+|macOS on ARM |arm64 |argocd-macos-arm64.tar.gz
```
```suggestion
|macOS on Intel |x86_64 |`argocd-macos-amd64.tar.gz`
|macOS on ARM |arm64 |`argocd-macos-arm64.tar.gz`
```

adellape commented on modules/gitops-installing-argocd-cli-on-macos.adoc:
```diff
@@ -10,29 +10,42 @@ For macOS, you can download the {gitops-shortname} `argocd` CLI as a `tar.gz` ar
-You can also download any version of the {gitops-shortname} `argocd` CLI by navigating to the corresponding directory for the version that you want in the link:https://mirror.openshift.com/pub/openshift-v4/clients/argocd-cli/[`argocd` CLI client download mirror].
+[NOTE]
+====
+Newer versions of the CLI tool is compatible with the older versions of {gitops-title} server but not vice-versa.
```
```suggestion
Newer versions of the CLI tool are compatible with the older versions of {gitops-title} server, but not vice versa.
```

adellape commented on modules/gitops-installing-argocd-cli-on-windows.adoc:
```diff
@@ -10,14 +10,23 @@ For Windows, you can download the {gitops-shortname} `argocd` CLI as a compresse
+|===
+|Operating system |Architecture |Tarball
 
+|Windows |x86_64 |argocd-windows-amd64.zip
```
```suggestion
|Windows |x86_64 |`argocd-windows-amd64.zip`
```

adellape commented on modules/gitops-installing-argocd-cli-on-windows.adoc:
```diff
@@ -10,14 +10,23 @@ For Windows, you can download the {gitops-shortname} `argocd` CLI as a compresse
-You can also download any version of the {gitops-shortname} `argocd` CLI by navigating to the corresponding directory for the version that you want in the link:https://mirror.openshift.com/pub/openshift-v4/clients/argocd-cli/[`argocd` CLI client download mirror].
+[NOTE]
+====
+Newer versions of the CLI tool is compatible with the older versions of {gitops-title} server but not vice-versa.
```
```suggestion
Newer versions of the CLI tool are compatible with the older versions of {gitops-title} server, but not vice versa.
```

Srivaralakshmi commented:
Thanks @adellape for the review comments. I have addressed them all and squashed the commits too. PTAL at it and merge the PR if everything checks okay.

adellape commented:
/cherrypick gitops-docs-1.12

Srivaralakshmi commented:
> @Srivaralakshmi Should this be picked to `gitops-docs-1.13` as well or no?

@adellape Yes, CP to gitops-docs-1.13 too please! Thanks for asking :)

adellape commented:
@Srivaralakshmi As discussed on Slack, we're fine with the `gitops-docs-1.13` CP failing because that branch will be rebased from `gitops-docs-main` when 1.13 development begins in earnest.
